api/v1beta1/azuremachine_default.go

```diff
@@ -79,11 +83,17 @@ func (s *AzureMachineSpec) SetDataDisksDefaults() {
 }
 
 // SetIdentityDefaults sets the defaults for VM Identity.
-func (s *AzureMachineSpec) SetIdentityDefaults() {
+func (s *AzureMachineSpec) SetIdentityDefaults(subscriptionID string) {
 	if s.Identity == VMIdentitySystemAssigned {
 		if s.RoleAssignmentName == "" {
 			s.RoleAssignmentName = string(uuid.NewUUID())
 		}
+		if s.Scope == "" {
+			s.Scope = fmt.Sprintf("/subscriptions/%s/", subscriptionID)
```

Comment:
would it make sense to use the `azure.GenerateSubscriptionScope(<SubscriptionID>)` function here ( and in the azuremachinepool_default.go) and the `azure.GenerateContributorRoleDefinitionID(<SubscriptionID>)` for the RoleDefinition (from `azure/defaults.go` ) rather than redefining the same `printf` ? 

Alternatively I guess we can remove those functions from defaults since they are only used for this purpose ( at the moment only in the `azure/scope/machine.go` )

Reply:
It would make sense! However, we can't import from the `azure` package from `api`, so I'll think about either moving the functions or removing them entirely